<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>Command line - Vulture Tutorial</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Command line";
        var mkdocs_page_input_path = "1. Running on local computer/2_Commandline.md";
        var mkdocs_page_url = null;
      </script>
    
    <script src="../../js/jquery-3.6.0.min.js" defer></script>
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
      <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Vulture Tutorial
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../..">Welcome</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">1. Running on local computer</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../1_Quickstart/">Quick start</a>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">Command line</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#requirements">Requirements</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#general-usage">General usage</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#0-prerequiresits-to-download-genome-files">0. Prerequiresits to download genome files</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#1-map-10x-scrna-seq-reads-to-the-viral-microbial-host-reference-set">1. Map 10x scRNA-seq reads to the viral microbial host reference set:</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#2-filter-the-mapped-umis-using-emptydrops-to-get-the-viral-and-microbial-host-filtered-umi-counts-matrix-and-also-output-viral-genes-and-barcodes-info-files">2. Filter the mapped UMIs using EmptyDrops to get the viral (and microbial) host filtered UMI counts matrix and also output viral genes and barcodes info files:</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#3-optional-and-starsolo-or-cellranger-only-output-some-quality-control-criteria-of-the-post-emptydrops-viral-microbial-supporting-reads-in-the-bam-file">3. (Optional, and STARsolo or CellRanger only) Output some quality control criteria of the post-EmptyDrops viral microbial supporting reads in the BAM file</a>
    </li>
        </ul>
    </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../3_Docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../4_Nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">2. Running on AWS Cloud</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../2.%20Running%20on%20AWS%20Cloud/1_Setup/">Setup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../2.%20Running%20on%20AWS%20Cloud/2_Nextflow/">Nextflow</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../2.%20Running%20on%20AWS%20Cloud/3_Batch/">Batch</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../2.%20Running%20on%20AWS%20Cloud/4_RunVulture/">Running Vulture on AWS cloud with Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Supplementary</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../Supplementary/Launchtemp/">Launch template Setup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../Supplementary/Supplementary/">Supplementary</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Vulture Tutorial</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a> &raquo;</li>
          <li>1. Running on local computer &raquo;</li>
      <li>Command line</li>
    <li class="wy-breadcrumbs-aside">
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="create-batch-compute-environmnet">Create Batch Compute Environmnet</h1>
<h2 id="requirements"><a name="require"></a>Requirements</h2>
<ul>
<li>Input data: 10x Chromium scRNA-seq reads</li>
<li>DropletUtils &gt;= v1.10.2</li>
<li>STAR &gt;= v2.7.9a or</li>
<li>cellranger &gt;= 6.0.0 or</li>
<li>Kallisto/bustools &gt;= 0.25.1 or</li>
<li>salmon/alevin &gt;= v1.4.0</li>
</ul>
<h2 id="general-usage"><a name="gen_usages"></a>General usage</h2>
<p>Map 10x scRNA-seq reads to the viral (and microbial) host reference set using STARsolo, CellRanger, Kallisto|bustools, or Salmon|Alevin. </p>
<h3 id="0-prerequiresits-to-download-genome-files">0. Prerequiresits to download genome files</h3>
<p>You need to download virus genome, prokaryotes genome, combined genome and virus combined genome in the following link and save them in a folder as "vmh_genome_dir" to be used in the next step.
<a href="https://vulture-reference.s3.ap-east-1.amazonaws.com/human_host_viruses_microbes.viruSITE.NCBIprokaryotes.with_hg38.removed_amb_viral_exon.gtf">human_host_viruses_microbes.viruSITE.NCBIprokaryotes.with_hg38.removed_amb_viral_exon.gtf</a>
<a href="https://vulture-reference.s3.ap-east-1.amazonaws.com/human_host_viruses_microbes.viruSITE.NCBIprokaryotes.with_hg38.fa">human_host_viruses_microbes.viruSITE.NCBIprokaryotes.with_hg38.fa</a>
<a href="https://vulture-reference.s3.ap-east-1.amazonaws.com/human_host_viruses.viruSITE.with_hg38.removed_amb_viral_exon.gtf">human_host_viruses.viruSITE.with_hg38.removed_amb_viral_exon.gtf</a>
<a href="https://vulture-reference.s3.ap-east-1.amazonaws.com/human_host_viruses.viruSITE.with_hg38.fa">human_host_viruses.viruSITE.with_hg38.fa</a></p>
<h3 id="1-map-10x-scrna-seq-reads-to-the-viral-microbial-host-reference-set">1. Map 10x scRNA-seq reads to the viral microbial host reference set:</h3>
<pre><code class="language-sh">Usage: scvh_map_reads.pl [Options] &lt;vmh_genome_dir&gt; &lt;R2&gt; &lt;R1&gt; or &lt;vmh_genome_dir&gt; &lt;.bam file&gt;

Options:                                                                                                                                Defaults
-o/--output-dir &lt;string&gt;   the output directory                                                                                          [./]   
-t/--threads &lt;int&gt;         number of threads to run alignment with                                                                       [&lt;1&gt;]  
-d/--database &lt;string&gt;     select virus or virus and prokaryotes database, can be 'viruSITE' or 'viruSITE.NCBIprokaryotes'               [&lt;viruSITE.NCBIprokaryotes&gt;]
-e/--exe &lt;string&gt;          executable command or stand alone executable path of the alignment tool                                       [&lt;&gt;]
-s/--soloStrand &lt;string&gt;   STARsolo param: Reverse or Forward used for 10x 5' or 3' protocol, respectively                               [&lt;Reverse&gt;]
-w/--whitelist &lt;string&gt;    STARsolo param --soloCBwhitelist                                                                              [&lt;&quot;vmh_genome_dir&quot;/737K-august-2016.txt&gt;]
-r/--ram &lt;int&gt;             limitation of RAM usage. For STARsolo, param: limitGenomeGenerateRAM, limitBAMsortRAM unit by GB              [&lt;128&gt;]
-f/--soloFeature &lt;string&gt; STARsolo param:  See --soloFeatures in STARsolo manual                                                         [&lt;Gene&gt;]
-ot/--outSAMtype &lt;string&gt;  STARsolo param:  See --outSAMtype in STARsolo manual                                                          [&lt;BAM SortedByCoordinate&gt;]
-mm/--soloMultiMappers &lt;string&gt;  STARsolo param:  See --soloMultiMappers in STARsolo manual                                              [&lt;EM&gt;]
-a/--alignment &lt;string&gt;    Select alignment methods: 'STAR', 'KB', 'Alevin', or 'CellRanger'                                             [&lt;STAR&gt;]
-v/--technology &lt;string&gt;   KB param:  Single-cell technology used (`kb --list` to view)                                                  [&lt;10XV2&gt;]
--soloCBstart &lt;string&gt;  STARsolo param:  See --soloCBstart in STARsolo manual                                                            [&lt;1&gt;]
--soloCBlen &lt;string&gt;  STARsolo param:  See --soloCBlen in STARsolo manual                                                                [&lt;16&gt;]
--soloUMIstart &lt;string&gt;  STARsolo param:  See --soloUMIstart in STARsolo manual                                                          [&lt;17&gt;]
--soloUMIlen &lt;string&gt;  STARsolo param:  See --soloUMIlen in STARsolo manual                                                              [&lt;10&gt;]
--soloInputSAMattrBarcodeSeq &lt;string&gt;  STARsolo param:  See --soloInputSAMattrBarcodeSeq in STARsolo manual                              [&lt;CR UR&gt;]
</code></pre>
<p>For fastqs alignment option 'STAR', 'KB', and 'Alevin', run:</p>
<pre><code class="language-sh">perl scvh_map_reads.pl -t num_threads -o output_dir vmh_genome_dir R2.fastq.gz R1.fastq.gz
</code></pre>
<p>where <em>-t</em> is a user-specified integer indicating number of threads to run with, <em>output_dir</em> is a user-specified directory to place the outputs, <em>vmh_genome_dir</em> is a pre-generated viral (and microbial) host (human) reference set directory, <em>R2.fastq.gz</em> and <em>R1.fastq.gz</em> are input 10x scRNA-seq reads.</p>
<p>For fastqs option 'CellRanger', run:</p>
<pre><code class="language-sh">perl scvh_map_reads.pl -t num_threads -o output_dir vmh_genome_dir sample fastqs
</code></pre>
<p>where <em>sample</em> and <em>fastqs</em> are two cellranger arguments: <em>--sample</em> and <em>--fastqs</em>. See documentation in <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/count">cellranger count</a> to infer rules of fastq and sample naming.</p>
<p>For bam files, we only support STARsolo, run:</p>
<pre><code class="language-sh">perl scvh_map_reads.pl -t num_threads -o output_dir vmh_genome_dir your_bam_file.bam
</code></pre>
<h3 id="2-filter-the-mapped-umis-using-emptydrops-to-get-the-viral-and-microbial-host-filtered-umi-counts-matrix-and-also-output-viral-genes-and-barcodes-info-files">2. Filter the mapped UMIs using EmptyDrops to get the viral (and microbial) host filtered UMI counts matrix and also output viral genes and barcodes info files:</h3>
<pre><code class="language-sh">Usage: Rscript scvh_filter_matrix.r output_dir sample_name
</code></pre>
<p>where <em>sample_name</em> is an optional user-specified tag to be used as a prefix for the output files.</p>
<h3 id="3-optional-and-starsolo-or-cellranger-only-output-some-quality-control-criteria-of-the-post-emptydrops-viral-microbial-supporting-reads-in-the-bam-file">3. (Optional, and STARsolo or CellRanger only) Output some quality control criteria of the post-EmptyDrops viral microbial supporting reads in the BAM file</h3>
<pre><code class="language-sh">Usage: perl scvh_analyze_bam.pl output_dir sample_name
</code></pre>
<div class="code-example" markdown="1">
[Previous Step](https://juychen.github.io/docs/6_Local/Quickstart.html){: .btn }
[Next Step](https://juychen.github.io/docs/6_Local/Nextflow.html){: .btn .btn-purple }
</div>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../1_Quickstart/" class="btn btn-neutral float-left" title="Quick start"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../3_Docker/" class="btn btn-neutral float-right" title="Docker">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../1_Quickstart/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../3_Docker/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme_extra.js" defer></script>
    <script src="../../js/theme.js" defer></script>
      <script src="../../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
